app/assets/javascripts/darkswarm/services/matcher.js.coffee

```diff
@@ -1,7 +1,16 @@
 Darkswarm.factory "Matcher", ->
-  # Match text fragment in an array of strings.
   new class Matcher
+
+    # Match text fragment in an array of strings.
     match: (properties, text)->
       properties.some (prop)->
         prop ||= ""
         prop.toLowerCase().indexOf(text.toLowerCase()) != -1
+
+    # Return true if text occurs at the beginning of any word present in an array of strings
+    matchBeginning: (properties, text) ->
+      text = text.trim()
+      properties.some (prop) ->
+        prop ||= ""
```

Comment:
What about a regular expression here to do a case-insensitive search for ``text`` at the beginning of the prop name or following a whitespace character or dash (or whatever other characters we choose to specify). 

Something like (untested):
```
text = text.trim()
regexp = new RegExp("(?:^|[\\s-])#{text}", "i")
properties.some (prop) -> (prop || "").search(regexp) >= 0
```